src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -10170,5 +10170,78 @@ public void Utf8Addition()
                 """;
             CreateCompilation(code, targetFramework: TargetFramework.Net70).VerifyDiagnostics();
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75592")]
+        public void SelfAssignment_ReturnOnly()
+        {
+            var source = """
+                ref struct S
+                {
+                    int field;
+                    ref int refField;
+
+                    static void M(ref S s)
+                    {
+                        s.refField = ref s.field;
+                    }
+                }
+                """;
+            CreateCompilation(source, targetFramework: TargetFramework.Net70).VerifyDiagnostics(
+                // (8,9): error CS9079: Cannot ref-assign 's.field' to 'refField' because 's.field' can only escape the current method through a return statement.
+                //         s.refField = ref s.field;
+                Diagnostic(ErrorCode.ERR_RefAssignReturnOnly, "s.refField = ref s.field").WithArguments("refField", "s.field").WithLocation(8, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/75592")]
+        public void SelfAssignment_CallerContext()
+        {
+            var source = """
+                using System.Diagnostics.CodeAnalysis;
+
+                S s = default;
+                S.M(ref s);
```

Comment:
Consider that these two forms are equivalent 

```csharp
ref struct S {
  [UnscopedRef]
  void M1() { }

  static void M2([UnscopedRef] ref S p) { }
}

S local;
local.M1();
S.M2(ref local);
```

So if the `S.M(ref s)` is an error then virtually any instance method call on a `ref struct` with `[UnscopedRef]` is too..